sdk/client/src/blocking.rs

```diff
@@ -0,0 +1,238 @@
+    validate, validate_batch, BatchResponse,
+};
+use crate::{
+    views::{
+        AccountStateWithProofView, AccountView, CurrencyInfoView, EventView, MetadataView,
+        StateProofView, TransactionView,
+    },
+    Error, Result,
+};
+use diem_types::{account_address::AccountAddress, transaction::SignedTransaction};
+use serde::{de::DeserializeOwned, Serialize};
+
+const REQUEST_TIMEOUT: u64 = 10_000;
+
+#[derive(Clone, Debug)]
+pub struct BlockingClient {
+    url: String,
+}
+
+impl BlockingClient {
+    pub fn new<T: Into<String>>(url: T) -> Self {
+        Self { url: url.into() }
+    }
+
+    pub fn batch(
+        &self,
+        requests: Vec<MethodRequest>,
+    ) -> Result<Vec<Result<Response<MethodResponse>>>> {
+        self.send_batch(requests)
+    }
+
+    pub fn submit(&self, txn: &SignedTransaction) -> Result<Response<()>> {
+        self.send(MethodRequest::submit(txn).map_err(Error::request)?)
+    }
+
+    pub fn get_metadata_by_version(&self, version: u64) -> Result<Response<MetadataView>> {
+        self.send(MethodRequest::get_metadata_by_version(version))
+    }
+
+    pub fn get_metadata(&self) -> Result<Response<MetadataView>> {
+        self.send(MethodRequest::get_metadata())
+    }
+
+    pub fn get_account(&self, address: AccountAddress) -> Result<Response<Option<AccountView>>> {
+        self.send(MethodRequest::get_account(address))
+    }
+
+    pub fn get_transactions(
+        &self,
+        start_seq: u64,
+        limit: u64,
+        include_events: bool,
+    ) -> Result<Response<Vec<TransactionView>>> {
+        self.send(MethodRequest::get_transactions(
+            start_seq,
+            limit,
+            include_events,
+        ))
+    }
+
+    pub fn get_account_transaction(
+        &self,
+        address: AccountAddress,
+        seq: u64,
+        include_events: bool,
+    ) -> Result<Response<Option<TransactionView>>> {
+        self.send(MethodRequest::get_account_transaction(
+            address,
+            seq,
+            include_events,
+        ))
+    }
+
+    pub fn get_account_transactions(
+        &self,
+        address: AccountAddress,
+        start_seq: u64,
+        limit: u64,
+        include_events: bool,
+    ) -> Result<Response<Vec<TransactionView>>> {
+        self.send(MethodRequest::get_account_transactions(
+            address,
+            start_seq,
+            limit,
+            include_events,
+        ))
+    }
+
+    pub fn get_events(
+        &self,
+        key: &str,
+        start_seq: u64,
+        limit: u64,
+    ) -> Result<Response<Vec<EventView>>> {
+        self.send(MethodRequest::get_events(key, start_seq, limit))
+    }
+
+    pub fn get_currencies(&self) -> Result<Response<Vec<CurrencyInfoView>>> {
+        self.send(MethodRequest::get_currencies())
+    }
+
+    pub fn get_network_status(&self) -> Result<Response<u64>> {
+        self.send(MethodRequest::get_network_status())
+    }
+
+    //
+    // Experimental APIs
```

Comment:
I wonder to what extent it might make sense for us to use traits and provide an interface for the stable API as well as the experimental API? That way, when we need to change the experimental API we don't have to worry about any side affects/changes to stable?

Reply:
These are marked as experimental simply because that's what the JsonRPC interface from the node marks them as. I agree that we do need nail down if these really do intend to change or not though.